### PR TITLE
Refactor Formik Proxy as Render Props enhancer

### PR DESCRIPTION
**Pull request by RafaelVidaurre**
**Please make sure the following boxes are checked before submitting your Pull Request:**

- [X] I've added this PR's link to its Asana task(s)
- [X] If this PR adds new code that is not going to change in the near future, it includes unit tests to cover it.

# This PR

- Transforms the Formik Proxy into a Render Props component to fix typing issues
- Fixes random typing issues on the FormItem.Input component
- Modifies all inputs to use the new version of the Formik Proxy enhancer

### CODE REVIEW

File: packages/new-polymath-issuer/src/pages/SecurityTokensDividends/Presenter.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React, { FC, Fragment, useState, useCallback } from 'react';
 import { utils, formatters, types } from '@polymathnetwork/new-shared';
 import * as Yup from 'yup';
-import { validateYupSchema, yupToFormErrors } from 'formik';
+import { validateYupSchema, yupToFormErrors, FastField } from 'formik';
```
Comment (monitz87) [inline packages/new-polymath-issuer/src/pages/SecurityTokensDividends/Presenter.tsx]: Why is the `FastField` import necessary?

File: packages/new-polymath-issuer/src/pages/SecurityTokensDividends/Presenter.tsx [hunk 2]
```diff
@@ -190,10 +190,7 @@ export const Presenter: FC<Props> = ({
             <Grid>
               <FormItem name="walletAddress">
                 <FormItem.Label>Wallet Address</FormItem.Label>
-                <FormItem.Input
-                  component={TextInput}
-                  placeholder="Wallet address"
-                />
+                <FormItem.Input component={TextInput} />
```
Comment (monitz87) [inline packages/new-polymath-issuer/src/pages/SecurityTokensDividends/Presenter.tsx]: Why remove the placeholder?

File: packages/new-polymath-ui/src/components/inputs/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -58,4 +58,16 @@ export const CheckboxPrimitive: FC<Props> = ({
   );
 };
 
-export const Checkbox = formikProxy(CheckboxPrimitive);
+const EnhancedCheckbox: FC<FormikExternalProps> = ({
+  field,
+  form,
+  ...rest
+}) => (
+  <FormikProxy<Value>
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/inputs/Checkbox/Checkbox.tsx]: `Value` isn't defined anywhere

File: packages/new-polymath-ui/src/components/inputs/Select/Select.tsx [hunk 4]
```diff
@@ -149,7 +149,13 @@ export const SelectPrimitive = Object.assign(ThemedSelectPrimitive, {
   defaultProps: SelectPrimitiveBase.defaultProps,
 });
 
-const EnhancedSelect = formikProxy(SelectPrimitive);
+const EnhancedSelect: FC<FormikExternalProps> = ({ field, form, ...rest }) => (
+  <FormikProxy<any>
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/inputs/Select/Select.tsx]: Is this OK having `any` as the value type?

File: packages/new-polymath-ui/src/components/inputs/TextInput/TextInput.tsx [hunk 5]
```diff
@@ -34,7 +37,18 @@ TextInputPrimitive.defaultProps = {
   value: '',
 };
 
-const EnhancedTextInput = formikProxy(TextInputPrimitive);
-export const TextInput = Object.assign(EnhancedTextInput, {
+const EnhancedDatePickerInput: FC<FormikExternalProps> = ({
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/inputs/TextInput/TextInput.tsx]: Shouldn't this be called `EnhancedTextInput`?

File: packages/new-polymath-ui/src/components/inputs/TimePickerSelect/TimePickerSelect.tsx [hunk 6]
```diff
@@ -66,4 +75,25 @@ class TimePickerSelectBase extends Component<TimePickerSelectProps> {
   }
 }
 
-export const TimePickerSelect = formikProxy(TimePickerSelectBase);
+// export const TimePickerSelect = formikProxy(TimePickerSelectBase);
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/inputs/TimePickerSelect/TimePickerSelect.tsx]:
```suggestion
```

File: packages/new-polymath-ui/src/components/inputs/TimePickerSelect/TimePickerSelect.tsx [hunk 7]
```diff
@@ -1,15 +1,24 @@
...
+import {
+  FormikProxy,
+  FormikExternalProps,
+} from '~/components/inputs/FormikProxy';
 
 export interface TimePickerSelectProps extends SelectProps {
   format: string;
   onChange: (value: number) => void;
   onBlur: () => void;
   value: any;
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/inputs/TimePickerSelect/TimePickerSelect.tsx]: Shouldn't this be typed as `number`?

File: packages/new-polymath-ui/src/components/inputs/TimePickerSelect/TimePickerSelect.tsx [hunk 8]
```diff
@@ -66,4 +75,25 @@ class TimePickerSelectBase extends Component<TimePickerSelectProps> {
...
 
-export const TimePickerSelect = formikProxy(TimePickerSelectBase);
+// export const TimePickerSelect = formikProxy(TimePickerSelectBase);
+
+const EnhancedTimePickerSelect: FC<ExternalProps> = ({
+  field,
+  form,
+  ...rest
+}) => (
+  <FormikProxy<Value>
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/inputs/TimePickerSelect/TimePickerSelect.tsx]: I don't see `Value` defined anywhere

File: packages/new-polymath-ui/src/components/inputs/ToggleInput/index.tsx [hunk 9]
```diff
@@ -140,4 +142,12 @@ export class ToggleInputPrimitive extends Component<ToggleInputProps> {
   }
 }
 
-export const ToggleInput = formikProxy(ToggleInputPrimitive);
+const EnhancedToggleInput: FC<ExternalProps> = ({ field, form, ...rest }) => (
+  <FormikProxy<Value>
```
Comment (monitz87) [inline packages/new-polymath-ui/src/components/inputs/ToggleInput/index.tsx]: I don't see `Value` defined anywhere

File: packages/new-polymath-issuer/src/pages/SecurityTokensDividends/Presenter.tsx (hunk 1 above)
Comment (RafaelVidaurre) [reply packages/new-polymath-issuer/src/pages/SecurityTokensDividends/Presenter.tsx]: Whoops!

File: packages/new-polymath-ui/src/components/inputs/Checkbox/Checkbox.tsx (hunk 3 above)
Comment (RafaelVidaurre) [reply packages/new-polymath-ui/src/components/inputs/Checkbox/Checkbox.tsx]: interesting, how did this not get signaled by the editor

File: packages/new-polymath-issuer/src/pages/SecurityTokensDividends/Presenter.tsx (hunk 2 above)
Comment (RafaelVidaurre) [reply packages/new-polymath-issuer/src/pages/SecurityTokensDividends/Presenter.tsx]: Did this without noticing, adding it back

File: packages/new-polymath-ui/src/components/inputs/Select/Select.tsx (hunk 4 above)
Comment (RafaelVidaurre) [reply packages/new-polymath-ui/src/components/inputs/Select/Select.tsx]: It is fine in this scenario. I don't see other way to type this part tbh

File: packages/new-polymath-ui/src/components/inputs/TextInput/TextInput.tsx (hunk 5 above)
Comment (RafaelVidaurre) [reply packages/new-polymath-ui/src/components/inputs/TextInput/TextInput.tsx]: Whoops

File: packages/new-polymath-ui/src/components/inputs/TimePickerSelect/TimePickerSelect.tsx (hunk 6 above)
Comment (RafaelVidaurre) [reply packages/new-polymath-ui/src/components/inputs/TimePickerSelect/TimePickerSelect.tsx]: 👍

File: packages/new-polymath-ui/src/components/inputs/TimePickerSelect/TimePickerSelect.tsx (hunk 7 above)
Comment (RafaelVidaurre) [reply packages/new-polymath-ui/src/components/inputs/TimePickerSelect/TimePickerSelect.tsx]: No idea, not my typing, I'll try changing it but I'm worried this cascades a set of type issues, we'll see

**Pull request merged**